cpp/src/arrow/util/decimal.cc

```diff
@@ -23,12 +23,55 @@
 #include <limits>
 #include <sstream>
 
+#include <boost/regex.hpp>
+
 #include "arrow/util/bit-util.h"
 #include "arrow/util/decimal.h"
 #include "arrow/util/logging.h"
 
 namespace arrow {
 
+static const Decimal128 ScaleMultipliers[] = {
+    Decimal128(0LL),
+    Decimal128(10LL),
+    Decimal128(100LL),
+    Decimal128(1000LL),
+    Decimal128(10000LL),
+    Decimal128(100000LL),
+    Decimal128(1000000LL),
+    Decimal128(10000000LL),
+    Decimal128(100000000LL),
+    Decimal128(1000000000LL),
+    Decimal128(10000000000LL),
+    Decimal128(100000000000LL),
+    Decimal128(1000000000000LL),
+    Decimal128(10000000000000LL),
+    Decimal128(100000000000000LL),
+    Decimal128(1000000000000000LL),
+    Decimal128(10000000000000000LL),
+    Decimal128(100000000000000000LL),
+    Decimal128(1000000000000000000LL),
+    Decimal128(0LL, 10000000000000000000ULL),
+    Decimal128(5LL, 7766279631452241920ULL),
+    Decimal128(54LL, 3875820019684212736ULL),
+    Decimal128(542LL, 1864712049423024128ULL),
+    Decimal128(5421LL, 200376420520689664ULL),
+    Decimal128(54210LL, 2003764205206896640ULL),
+    Decimal128(542101LL, 1590897978359414784ULL),
+    Decimal128(5421010LL, 15908979783594147840ULL),
+    Decimal128(54210108LL, 11515845246265065472ULL),
+    Decimal128(542101086LL, 4477988020393345024ULL),
+    Decimal128(5421010862LL, 7886392056514347008ULL),
+    Decimal128(54210108624LL, 5076944270305263616ULL),
+    Decimal128(542101086242LL, 13875954555633532928ULL),
+    Decimal128(5421010862427LL, 9632337040368467968ULL),
+    Decimal128(54210108624275LL, 4089650035136921600ULL),
+    Decimal128(542101086242752LL, 4003012203950112768ULL),
+    Decimal128(5421010862427522LL, 3136633892082024448ULL),
+    Decimal128(54210108624275221LL, 12919594847110692864ULL),
+    Decimal128(542101086242752217LL, 68739955140067328ULL),
+    Decimal128(5421010862427522170LL, 687399551400673280ULL)};
```

Comment:
how were these generated, just for future reference?

Reply:
By looping over the old ones and printing out `value.high_bits(), value.low_bits()`